### ISSUE TITLE
Issue loading PDF files on windows 11

### ISSUE THREAD

**Issue by JEHollierJr** [llmware-ai/llmware#159]
I am having problems with loading some of my PDF files on Windows 11 Pro.

Your examples seem to work fine for parsing PDF files, but when I use some of my own PDF files, I get the error below.
The file is 50 pages with tables and graphics.

Traceback (most recent call last):
  File "d:\home\john\Projects\llmware-Project\LLMWareCreateLib.py", line 29, in <module>
    library.add_files(os.path.join(sample_files_path,File_Dir))
  File "C:\Users\jehol\AppData\Local\Programs\Python\Python311\Lib\site-packages\llmware\library.py", line 357, in add_files    parsing_results = Parser(library=self).ingest(input_folder_path,dupe_check=True)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jehol\AppData\Local\Programs\Python\Python311\Lib\site-packages\llmware\parsers.py", line 358, in ingest    
    self.parse_pdf(self.pdf_work_folder, save_history=False)
  File "C:\Users\jehol\AppData\Local\Programs\Python\Python311\Lib\site-packages\llmware\parsers.py", line 719, in parse_pdf    pages_created = pdf_handler(account_name, library_name, fp_c, collection_db_path_c, image_fp_c,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: exception: access violation reading 0x0000023068F20360
PS D:\home\john\Projects\llmware-Project>

Can you help me work through this?  Let me know if I can provide more information.

Thanks,
John

**Comment by gongwork** [comment]
Hi @doberst  
I encounter similar issue (not the same) when loading 1 pdf file, 
Here is the error trace:
```
> Adding some files to the library...
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_6980\4202083958.py in <module>
      1 print (f"\n > Adding some files to the library...")
----> 2 library.add_files(input_folder_path=os.path.join(sample_files_path, "SmallLibrary"))

~\anaconda3\lib\site-packages\llmware\library.py in add_files(self, input_folder_path)
    355         logging.info("update: lib_counters_before - %s ", lib_counters_before)
    356 
--> 357         parsing_results = Parser(library=self).ingest(input_folder_path,dupe_check=True)
    358 
    359         logging.info("update: parsing results - %s ", parsing_results)

~\anaconda3\lib\site-packages\llmware\parsers.py in ingest(self, input_folder_path, dupe_check)
    356 
    357         if work_order["pdf"] > 0:
--> 358             self.parse_pdf(self.pdf_work_folder, save_history=False)
    359             self.uploads(self.pdf_work_folder)
    360 

~\anaconda3\lib\site-packages\llmware\parsers.py in parse_pdf(self, fp, write_to_db, save_history, image_save)
    717         #       char * db_pw
    718 
--> 719         pages_created = pdf_handler(account_name, library_name, fp_c, collection_db_path_c, image_fp_c,
    720                                     input_debug_mode, input_image_save_mode, write_to_db_on_c,
    721                                     write_to_filename_c, user_block_size, unique_doc_num_c,

OSError: exception: access violation writing 0x00000200D0998576
```

By adding pdf 1-by-1, I was able to pin down which pdf file causing error.

However, I suggest, parser.py:ingest() should add error-handling to ignore bad file and print out the filename, and continue processing.

As I start learning `llmware` today, I find it helpful to try example in jupyter notebook, which has better error trace.

Thanks,
Wen

**Comment by doberst** [comment]
Hi John and Wen - really appreciate you both highlighting this issue.  We were able to reproduce the problem independently as well.   We have updated the Windows PDF parser in the new PyPy release yesterday (0.1.11) - including improvements in memory management.  We did identify a couple of areas where the Windows compiler was more sensitive than Linux and Mac and have updated the code.    Could you pull 0.1.11, or alternatively the latest in the repo, and let us know if this fixes the issue?

**Comment by gongwork** [comment]
@doberst  yes, it is fixed in `__version__ = '0.1.11.4'`, Thank you

**Comment by JEHollierJr** [comment]
Yes, I verified that my files load now in version 0.1.11.4.  Thank you.
